x/axelarnet/keeper/coin.go

```diff
@@ -0,0 +1,155 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibctypes "github.com/cosmos/ibc-go/v2/modules/apps/transfer/types"
+
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/exported"
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/types"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+type coin struct {
+	sdk.Coin
+	ctx      sdk.Context
+	k        Keeper
+	nexusK   types.Nexus
+	coinType types.CoinType
+}
+
+func newCoin(ctx sdk.Context, k Keeper, nexusK types.Nexus, c sdk.Coin) coin {
+	return coin{
+		Coin:     c,
+		ctx:      ctx,
+		k:        k,
+		nexusK:   nexusK,
+		coinType: getCoinType(ctx, nexusK, c.Denom),
+	}
+}
+
+// lock locks coin from deposit address to escrow address
+func (c coin) lock(ibcK IBCKeeper, bankK types.BankKeeper, depositAddr sdk.AccAddress) error {
```

Comment:
```suggestion
func (c coin) Lock(ibcK IBCKeeper, bankK types.BankKeeper, depositAddr sdk.AccAddress) error {
```
To make it clear what the interface is?